earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/callbacks.h

```diff
@@ -8,4 +8,6 @@ class IndexItem;
 
 using map_packet_walker = std::function<bool(int layer, const IndexItem&)>;
 
+using map_file_walker = std::function<bool(const std::string&)>;

```

Comment:
Add `int layer` to the callback so those walking all layers know what layer the file.  It is redundant for those calling the single layer walker.  If that bothers you you could have two walker callback types one for each of the file walking methods.  Or if it doesn't bother you I think it is fine to be redundant and use the same walker callback for both.

Reply:
That doesn't bother me, seems right that you'd want to easily know what layer something is on and the redundancy created is pretty minor and only for one use case.